src/data/media-player.ts

```diff
@@ -17,30 +15,47 @@ export const SUPPORT_STOP = 4096;
 export const SUPPORTS_PLAY = 16384;
 export const SUPPORT_SELECT_SOUND_MODE = 65536;
 export const OFF_STATES = ["off", "idle"];
+export const CONTRAST_RATIO = 3.5;
 
 export interface MediaPlayerThumbnail {
   content_type: string;
   content: string;
 }
 
-export const fetchMediaPlayerThumbnailWithCache = (
-  hass: HomeAssistant,
-  entityId: string
-) =>
-  timeCachePromiseFunc(
-    "_media_playerTmb",
-    9000,
-    fetchMediaPlayerThumbnail,
-    hass,
-    entityId
-  );
+export const getCurrentProgress = (stateObj: HassEntity): number => {
+  let progress = stateObj.attributes.media_position;
+  progress +=
+    (Date.now() -
+      new Date(stateObj.attributes.media_position_updated_at).getTime()) /
+    1000.0;
+  return progress;
+};
+
+export const computeSecondaryTitle = (stateObj: HassEntity): string => {
```

Comment:
This seems like something that should remain in the card file as it is very much related to where it is used (it even has secondary in the name). You can still put it above the class definition there.

Reply:
We can use this same function for the media row

Reply:
https://github.com/home-assistant/frontend/blob/dev/src/panels/lovelace/entity-rows/hui-media-player-entity-row.ts#L69

Reply:
Okay then rename the function . computeMediaDescription?